### PR TITLE
Doc/auto api

### PR DESCRIPTION
**Pull request by weidel-p**
Issue Number: 

Objective of pull request:

This PR changes the format of several docstrings to allow shpinx to generate documentation.

## Pull request checklist

Your PR fulfills the following requirements:
- [ ] [Issue](https://github.com/lava-nc/lava/issues) created that explains the change and why it's needed
- [ ] Tests are part of the PR (for bug fixes / features)
- [ ] [Docs](https://github.com/lava-nc/docs) reviewed and added / updated if needed (for bug fixes / features)
- [ ] PR conforms to [Coding Conventions](https://lava-nc.org/developer_guide.html#coding-conventions)
- [ ] [PR applys BSD 3-clause or LGPL2.1+ Licenses](https://lava-nc.org/developer_guide.html#add-a-license) to all code files
- [x] Lint (`flakeheaven lint src/lava tests/`) and (`bandit -r src/lava/.`) pass locally
- [ ] Build tests (`pytest`) passes locally

## Pull request type

 

 

Please check your PR type:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Documentation changes
- [ ] Other (please describe): 

## What is the current behavior?

-

## What is the new behavior?

-

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/lava/magma/core/decorator.py [hunk 1]
```diff
@@ -130,13 +130,13 @@ def tag(*args: ty.Union[str, ty.List[str]]):
...
-    @tag('bit-accurate', 'loihi')
-    class ExampleProcModel(AbstractProcessModel):...
-        -> these tags identify a particular ProcessModel as being
-        bit-accurate with Loihi hardware platform. Which means,
-        the numerical output produced by such a ProcessModel on a CPU would be
-        exactly same as that of Loihi.
+    >>> @implements(proc=ExampleProcess)
+    >>> @tag('bit-accurate', 'loihi')
+    >>> class ExampleProcModel(AbstractProcessModel):...
+    these tags identify a particular ProcessModel as being
```
Comment (mathisrichter) [inline src/lava/magma/core/decorator.py:136]: -> "These tags ..."

File: src/lava/magma/core/decorator.py [hunk 2]
```diff
@@ -130,13 +130,13 @@ def tag(*args: ty.Union[str, ty.List[str]]):
...
+    bit-accurate with Loihi hardware platform. Which means,
```
Comment (mathisrichter) [inline src/lava/magma/core/decorator.py:137]: "Which means, ..." -> "This means that..."

File: src/lava/magma/core/decorator.py [hunk 3]
```diff
@@ -130,13 +130,13 @@ def tag(*args: ty.Union[str, ty.List[str]]):
...
+    the numerical output produced by such a ProcessModel on a CPU would be
+    exactly same as that of Loihi.
```
Comment (mathisrichter) [inline src/lava/magma/core/decorator.py:139]: "exactly same" -> "the same as on Loihi."

File: src/lava/magma/core/process/process.py [hunk 4]
```diff
@@ -393,6 +397,7 @@ def load(self, path: str):
 
     @property
     def model(self) -> "AbstractProcessModel":
+        """ Return model """
```
Comment (mathisrichter) [inline src/lava/magma/core/process/process.py:400]: See my comment in the other PR.

File: src/lava/magma/core/run_configs.py [hunk 5]
```diff
@@ -140,8 +139,8 @@ class AbstractLoihiRunCfg(RunConfig):
         dict are respected over any logic. For example, {Dense: PyDenseModel}.
         Note that this is a dict mapping classnames to classnames.
     loglevel: int
-              sets level of event logging, as defined by Python's 'logging'
-              facility. Default: logging.WARNING
+        sets level of event logging, as defined by Python's 'logging'
```
Comment (mathisrichter) [inline src/lava/magma/core/run_configs.py:142]: Sets...

File: src/lava/magma/core/sync/protocol.py [hunk 6]
```diff
@@ -15,18 +15,19 @@ class AbstractSyncProtocol:
...
 
-    The phases of execution and synchronizations points are implemented in the
-    specific `RuntimeService`.
+    The phases of execution and synchronizations points are implemented in
+    the specific `RuntimeService`.
     """
     @property
     @abstractmethod
     def runtime_service(self):
+        """ Not implemented """
```
Comment (mathisrichter) [inline src/lava/magma/core/sync/protocol.py:32]: Should this be throwing a `NotImplementedError`?

File: src/lava/proc/sdn/process.py [hunk 7]
```diff
@@ -13,14 +14,14 @@
 @unique
 class ActivationMode(IntEnum):
     """Enum for synapse sigma delta activation mode. Options are
-    {``UNIT : 0``, ``RELU : 1``}.
+    UNIT: 0
```
Comment (mathisrichter) [inline src/lava/proc/sdn/process.py:17]: Remove the redundant listing of the options in the docstring.

**Review by mathisrichter — APPROVED** [review]
Nice! Again, please consider removing the docstrings for trivial methods.

File: src/lava/proc/sdn/process.py (hunk 7 above)
Comment (weidel-p) [reply src/lava/proc/sdn/process.py:17]: What do you mean?

File: src/lava/magma/core/process/variable.py [hunk 8]
```diff
@@ -63,6 +64,7 @@ def __init__(
 
     @property
     def model(self):
+        """ Return model """
```
Comment (PhilippPlank) [inline src/lava/magma/core/process/variable.py:67]: """Return model.""" would look nicer.

File: src/lava/magma/core/sync/protocols/loihi_protocol.py [hunk 9]
```diff
@@ -117,6 +126,7 @@ def run_post_mgmt(self) -> None:
     # Synchronizer classes that implement protocol in a domain
     @property
     def runtime_service(self):
+        """ Return RuntimeService """
```
Comment (PhilippPlank) [inline src/lava/magma/core/sync/protocols/loihi_protocol.py:129]: Same here, end with full stop and remove extra spaces.

File: src/lava/magma/runtime/message_infrastructure/multiprocessing.py [hunk 10]
```diff
@@ -55,6 +55,7 @@ def run(self):
 
     @property
     def exception(self):
+        """ Exception property """
```
Comment (PhilippPlank) [inline src/lava/magma/runtime/message_infrastructure/multiprocessing.py:58]: Same.

File: src/lava/magma/runtime/message_infrastructure/nx.py [hunk 11]
```diff
@@ -33,7 +33,7 @@ def stop(self):
         """Stops the shared memory manager"""
```
Comment (PhilippPlank) [inline src/lava/magma/runtime/message_infrastructure/nx.py:33]: Full stop.

File: src/lava/magma/runtime/message_infrastructure/nx.py [hunk 12]
```diff
@@ -33,7 +33,7 @@ def stop(self):
...
         pass
 
-    def channel_class(self, channel_type: ChannelType) -> ty.Type[Channel]:
+    def channel_class(self, channel_type: ChannelType) -> ty.Type[ChannelType]:
         """Given a channel type, returns the shared memory based class
```
Comment (PhilippPlank) [inline src/lava/magma/runtime/message_infrastructure/nx.py:37]: Full stop missing.

File: src/lava/proc/sdn/process.py [hunk 13]
```diff
@@ -45,6 +47,7 @@ def __init__(
 
     @property
     def shape(self) -> ty.Tuple[int, ...]:
+        """ Return shape of the Process """
```
Comment (PhilippPlank) [inline src/lava/proc/sdn/process.py:50]: Same.

File: src/lava/proc/sdn/process.py [hunk 14]
```diff
@@ -110,6 +113,7 @@ def __init__(self,
...
```
Comment (PhilippPlank) [inline src/lava/proc/sdn/process.py:116]: Same.

File: src/lava/proc/sdn/process.py [hunk 15]
```diff
@@ -190,4 +194,5 @@ def __init__(
...
```
Comment (PhilippPlank) [inline src/lava/proc/sdn/process.py:197]: Same.

File: src/lava/magma/compiler/builders/interfaces.py [hunk 16]
```diff
@@ -55,11 +55,12 @@ class AbstractProcessBuilder(AbstractBuilder):
     Process builders instantiate and initialize a ProcessModel.
 
     Parameters
-        ----------
-        proc_model: AbstractProcessModel
-                    ProcessModel class of the process to build.
-        model_id: int
-                  model_id represents the ProcessModel ID to build.
+    ----------
```
Comment (PhilippPlank) [inline src/lava/magma/compiler/builders/interfaces.py:58]: Full stop missing in ln 37, 43, 48 in this file (cannot comment there directly).

**Review by PhilippPlank — APPROVED** [review]
Looks good. One general remark. I would suggest that each docstring is a sentence which ends with a full stop - similar to our inline comment rules.
I added a comment where I found missing full stops/sentences.

**Pull request merged**